CMakeLists.txt

```diff
@@ -19,7 +19,7 @@ project(diskann)
 
 cmake_minimum_required(VERSION 3.15)
 
-set(CMAKE_STANDARD 14)
+set(CMAKE_STANDARD 17)
```

Comment:
Does `CMAKE_STANDARD` even do anything? I couldn't find any documentation on it and I don't see any references in DiskANN itself.

From a search, it appears that the way to deal with C++ standard is to set `CMAKE_CXX_STANDARD` instead. If I'm reading it correctly, it should then automatically add the right compiler settings for the used compiler (`e.g. /std:c++17` for MSVC) rather than adding it explicitly like in the existing `CMakeLists.txt` files.

Reply:
Yes, I had that on my local, but because it compiled fine, i thought may be **CMAKE_STANDARD** does the trick.